be/src/olap/rowset/segment_creator.cpp

```diff
@@ -70,12 +71,14 @@ Status SegmentFlusher::flush_single_block(const vectorized::Block* block, int32_
         _context.tablet_schema->cluster_key_idxes().empty()) {
         std::unique_ptr<segment_v2::VerticalSegmentWriter> writer;
         RETURN_IF_ERROR(_create_segment_writer(writer, segment_id, no_compression));
-        RETURN_IF_ERROR(_add_rows(writer, &flush_block, 0, flush_block.rows()));
+        HANDLE_EXCEPTION_IF_CATCH_EXCEPTION(_add_rows(writer, &flush_block, 0, flush_block.rows()),
+                                            [&](const doris::Exception& ex) {});
         RETURN_IF_ERROR(_flush_segment_writer(writer, writer->flush_schema(), flush_size));
     } else {
         std::unique_ptr<segment_v2::SegmentWriter> writer;
         RETURN_IF_ERROR(_create_segment_writer(writer, segment_id, no_compression));
-        RETURN_IF_ERROR(_add_rows(writer, &flush_block, 0, flush_block.rows()));
+        HANDLE_EXCEPTION_IF_CATCH_EXCEPTION(_add_rows(writer, &flush_block, 0, flush_block.rows()),
```

Comment:
RETURN_IF_CATCH_EXCEPTION

Reply:
we should use RETURN_IF_ERROR_OR_CATCH_EXCEPTION, _add_rows will return status